test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -263,26 +275,30 @@ TEST_F(PickFirstTest, WithShuffle) {
   // prevent flake test.
   constexpr size_t kMaxTries = 10;
   std::vector<absl::string_view> addresses_after_update;
-  for (size_t i = 0; i < kMaxTries; i++) {
+  bool shuffled = false;
+  for (size_t i = 0; i < kMaxTries; ++i) {
     absl::Status status = ApplyUpdate(
         BuildUpdate(kAddresses, MakePickFirstConfig(true)), lb_policy_.get());
+    work_serializer_->Run([&]() { lb_policy_->ExitIdleLocked(); },
```

Comment:
Suggest moving this inside of `GetOrderAddressesArePicked()`, so that we don't need to duplicate it in 3 places.

Reply:
Done